packages/router-plugin/src/core/code-splitter/compilers.ts

```diff
@@ -652,6 +663,28 @@ export function compileCodeSplitVirtualRoute(
                   ),
                 ]),
               )
+            } else if (t.isBooleanLiteral(splitNode)) {
+              // Handle boolean literals (false/true)
+              if (splitNode.value === false) {
+                // false means "don't use this component, fallback to parent"
+                // Skip code splitting to preserve fallback behavior
+                return
+              }
+              // For true or other boolean values, proceed with splitting
```

Comment:
`true` is not a valid value here IMO
also, there are no "other boolean values" ...